jupyter_server_fileid/manager.py

```diff
@@ -95,6 +99,9 @@ def _from_normalized_path(self, path: Optional[str]) -> Optional[str]:
     def _move_recursive(self, old_path: str, new_path: str, path_mgr: Any = os.path) -> None:
         """Move all children of a given directory at `old_path` to a new
         directory at `new_path`, delimited by `sep`."""
+        if not hasattr(self, "con") or self.con is None:
```

Comment:
I don't think the attribute existence check is necessary (here and elsewhere).
```suggestion
        if self.con is None:
```

Reply:
Oh yeah, this got added to prevent mypy from complaining. The real fix is to move all these SQLite-specific methods and attributes into a separate mixin like you had suggested previously. I've created an issue to track this separately: https://github.com/jupyter-server/jupyter_server_fileid/issues/53

Reply:
> Oh yeah, this got added to prevent mypy from complaining. 

Could you please be more specific regarding your `mypy` comment?  I just removed the `not hasattr()` clauses and `pre-commit` (with `mypy`) ran successfully.  I'm just curious about what you're seeing.  Thanks.  I'm using `mypy-0.991`.

Reply:
Also, I think the `hasattr()` clause in `__del()__` needs to be extended with `and self.con is not None` prior to calling `self.con.commit()` and `self.con.close()` at L513.